### PR TITLE
SEQNG-1197 FIx for TCS in-position timeout problem.

### PR DESCRIPTION
**Pull request by jluhrs**
The main purpose of this PR is to fix the TCS in-position timeout problem. The fix was to avoid waiting for the in-position to stabilize when there was no offset applied. I found other minor problems that I fixed.
Other changes:

- Consolidated the construction of thread pools in the ACM module.
- Callback calls are now done in their own thread.

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsEpics.scala [hunk 1]
```diff
@@ -786,14 +810,19 @@ final class TcsEpicsImpl[F[_]: Async](epicsService: CaService, tops: Map[String,
   // for the in-position to change to true and stay true for stabilizationTime. It will wait up to `timeout`
   // seconds for that to happen.
   override def waitInPosition(stabilizationTime: Duration, timeout: FiniteDuration)(implicit T: Timer[F]): F[Unit] =
-    T.sleep(FiniteDuration(tcsSettleTime.toMillis, TimeUnit.MILLISECONDS)) *>
-      Sync[F].delay(filteredInPositionAttr.restart(stabilizationTime))
-        .flatMap(waitForValueF(_, "TRUE", timeout,"TCS inposition flag"))
+    T.sleep(FiniteDuration(tcsSettleTime.toMillis, TimeUnit.MILLISECONDS)) *> (
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsEpics.scala:813]: I guess this is the core fix

File: modules/seqexec/server/src/test/scala/seqexec/server/EpicsUtilSpec.scala [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import org.scalatest.flatspec.AnyFlatSpec
+import java.util.concurrent.TimeUnit.MILLISECONDS
+
+import scala.concurrent.duration.FiniteDuration
+
+class EpicsUtilSpec extends AnyFlatSpec {
+
+  val executor: ScheduledExecutorService = new ScheduledThreadPoolExecutor(2)
+
+  "EpicsUtil waitForValue" should "work with CaWindowStabilizer" in {
```
Comment (cquiroz) [inline modules/seqexec/server/src/test/scala/seqexec/server/EpicsUtilSpec.scala:21]: 👍

File: modules/acm/src/main/java/edu/gemini/epics/acm/CaWindowStabilizer.java [hunk 3]
```diff
@@ -28,13 +28,15 @@
     private final CaAttributeListener<T> valListener;
     private T filteredVal;
     private T lastVal;
+    private final AttributeNotifier<T> notifier;
 
-    public CaWindowStabilizer(CaAttribute<T> sa, Duration settleTime) {
+    public CaWindowStabilizer(CaAttribute<T> sa, Duration settleTime, ScheduledExecutorService executor) {
```
Comment (cquiroz) [inline modules/acm/src/main/java/edu/gemini/epics/acm/CaWindowStabilizer.java:33]: I don't recall this class from before 😧

File: modules/acm/src/main/java/edu/gemini/epics/acm/AttributeNotifier.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        }
+    }
+
+    synchronized public void removeListener(CaAttributeListener<T> listener) {
+        listeners.remove(listener);
+    }
+
+    public void notifyValueChangeSingle(CaAttributeListener<T> listener, List<T> vals) {
+        executor.execute(() -> {
+            synchronized (AttributeNotifier.this) {
```
[the remark this answers is not in the document]
Comment (jluhrs) [reply modules/acm/src/main/java/edu/gemini/epics/acm/AttributeNotifier.java:33]: Yes. This is one consequence of having the callback calls in their own threads. It open the possibility that the listener is removed after the callback call is scheduled, but before it is executed. That is why I added a check in the actual call.

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsEpics.scala (hunk 1 above)
Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsEpics.scala:813]: Yes it is.

File: modules/acm/src/main/java/edu/gemini/epics/acm/CaWindowStabilizer.java (hunk 3 above)
Comment (jluhrs) [reply modules/acm/src/main/java/edu/gemini/epics/acm/CaWindowStabilizer.java:33]: Which one, `CaWindowStabilizer`?

File: modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsEpics.scala (hunk 1 above)
Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsEpics.scala:813]: Yes and no. It is an improvement that would have fixed the problem, but the only case when this was called with a zero stabilization time was when there was no offset applied, and that case does not apply anymore.

**Pull request merged**